ddl/table.go

```diff
@@ -76,7 +76,17 @@ func onCreateTable(d *ddlCtx, t *meta.Meta, job *model.Job) (ver int64, _ error)
 		}
 		if atomic.LoadUint32(&EnableSplitTableRegion) != 0 {
 			// TODO: Add restrictions to this operation.
-			go splitTableRegion(d.store, tbInfo.ID)
+			pi := tbInfo.GetPartitionInfo()
+			if pi != nil {
+				// Max partition count is 4096, should we sample and just choose some of the partition to split?
+				go func(pi *model.PartitionInfo) {
```

Comment:
How about not `go`? Since the split process can be really slow. Client could start to insert data before split complete.

Reply:
I add a session variable `tidb_wait_table_split_finish` to control use sync or async split. Maybe we can use it. 
And because the split process can be really slow, I prefer to put the split process in ddl_api.go, not in the DDL owner to do this, otherwise sync split maybe block the DDL jobs.

Reply:
I've move the code from table.go to ddl_api.go